sdk/src/azure/core/az_http_request.c

```diff
@@ -97,6 +101,66 @@ AZ_NODISCARD az_result az_http_request_append_path(_az_http_request* ref_request
   return AZ_OK;
 }
 
+// returns the query parameter value from a query paramenter name.
+// Or returns AZ_SPAN_NULL if query parameter name is not found
+static AZ_NODISCARD az_span
+_az_http_request_find_query_parameter(_az_http_request* ref_request, az_span query_paramenter_name)
+{
+  if (ref_request->_internal.query_start == 0)
+  {
+    return AZ_SPAN_NULL;
+  }
+
+  az_span url = ref_request->_internal.url;
+  uint8_t* query_params_ptr = az_span_ptr(url);
+  uint8_t* new_qp_name_ptr = az_span_ptr(query_paramenter_name);
+  int32_t new_qp_name_size = az_span_size(query_paramenter_name);
+
+  // will be set to the start of the value of a qp equal to the new qp
+  int32_t value_index_start = -1;
+
+  for (int32_t is_query_start = 1, index = ref_request->_internal.query_start; // +1 to jump `?`
+       index < ref_request->_internal.url_length - new_qp_name_size;
+       index++)
```

Comment:
With the two loop-index variables, it becomes difficult to understand the logic of this loop. Can we extract out `is_query_start` into a local outside the loop?

Reply:
it is only used inside the loop. I am trying to follow a @JeffreyRichter  advise about avoiding to declare variables outside a loop if only required in loop. Then we know for sure that variable is never required again for anything

Reply:
> advise about avoiding to declare variables outside a loop if only required in loop

I think that is a good goal, but in this case, the variable essentially is outside the scope of the loop. We are just intentionally defining it in the for-loop definition, to try to force into that pattern, causing a bit of confusion.

If you re-write this using a while loop, you wouldn't be able to put `int32_t is_query_start` inside the loop body.
Either way, just a nit.